src/main/java/frc/trigon/robot/commands/factories/GeneralCommands.java

```diff
@@ -100,6 +106,16 @@ public static Command duplicate(Command command) {
         );
     }
 
+    public static Command getNoteCollectionCommand() {
+        if (RobotContainer.INTAKE.hasNote())
```

Comment:
This if will happen when the function is called, not when the command is scheduled.